### ISSUE TITLE
can't push repo via SSH

### ISSUE THREAD

**Issue by RanchoCooper** [sameersbn/docker-gitlab#806]
my company installed GitLab enviroment on Docker

but someone find out they can't push their repo via SSH

nothing wrong with ssh-pub-key, I already checked!

when they enter 

`git push -u origin master`

required to enter 

`gitlab@192.168.1.xxx's password: `

HTTPS works well

**Comment by mahzoun** [comment]
what is the output of ssh -v -p <ssh_port> git@<ssh_host> ?

**Comment by RanchoCooper** [comment]
Bad port

**Comment by mahzoun** [comment]
I updated my comment please try again and print the whole output.

**Comment by mike360** [comment]
I'm having the same problem -- actually, I can't clone from SSH URL, but the problem is probably the same. My account definitely has the SSH key added to it.

Here is my output from the above command:
`
$ ssh -v -p 30000 git@git.pickslide.tech
OpenSSH_6.9p1, LibreSSL 2.1.8
debug1: Reading configuration data /Users/Mike/.ssh/config
debug1: Reading configuration data /etc/ssh/ssh_config
debug1: /etc/ssh/ssh_config line 21: Applying options for *
debug1: Connecting to <hostname redacted> [<ip redacted>] port 30000.
debug1: Connection established.
debug1: identity file /Users/Mike/.ssh/id_rsa type 1
debug1: key_load_public: No such file or directory
debug1: identity file /Users/Mike/.ssh/id_rsa-cert type -1
debug1: key_load_public: No such file or directory
debug1: identity file /Users/Mike/.ssh/id_dsa type -1
debug1: key_load_public: No such file or directory
debug1: identity file /Users/Mike/.ssh/id_dsa-cert type -1
debug1: key_load_public: No such file or directory
debug1: identity file /Users/Mike/.ssh/id_ecdsa type -1
debug1: key_load_public: No such file or directory
debug1: identity file /Users/Mike/.ssh/id_ecdsa-cert type -1
debug1: key_load_public: No such file or directory
debug1: identity file /Users/Mike/.ssh/id_ed25519 type -1
debug1: key_load_public: No such file or directory
debug1: identity file /Users/Mike/.ssh/id_ed25519-cert type -1
debug1: Enabling compatibility mode for protocol 2.0
debug1: Local version string SSH-2.0-OpenSSH_6.9
debug1: Remote protocol version 2.0, remote software version OpenSSH_7.2p2 Ubuntu-4ubuntu1
debug1: match: OpenSSH_7.2p2 Ubuntu-4ubuntu1 pat OpenSSH* compat 0x04000000
debug1: Authenticating to <hostname redacted>:30000 as 'git'
debug1: SSH2_MSG_KEXINIT sent
debug1: SSH2_MSG_KEXINIT received
debug1: kex: server->client chacha20-poly1305@openssh.com <implicit> none
debug1: kex: client->server chacha20-poly1305@openssh.com <implicit> none
debug1: expecting SSH2_MSG_KEX_ECDH_REPLY
debug1: Server host key: ecdsa-sha2-nistp256 SHA256:8ie/lvNniy8xYjj9FGn3YIDcxvXtBb3NwH3Vh+KJQZ8
debug1: Host '[<hostname redacted>]:30000' is known and matches the ECDSA host key.
debug1: Found key in /Users/Mike/.ssh/known_hosts:147
debug1: SSH2_MSG_NEWKEYS sent
debug1: expecting SSH2_MSG_NEWKEYS
debug1: SSH2_MSG_NEWKEYS received
debug1: SSH2_MSG_SERVICE_REQUEST sent
debug1: SSH2_MSG_SERVICE_ACCEPT received
debug1: Authentications that can continue: publickey,password
debug1: Next authentication method: publickey
debug1: Offering RSA public key: /Users/Mike/.ssh/id_rsa
debug1: Authentications that can continue: publickey,password
debug1: Trying private key: /Users/Mike/.ssh/id_dsa
debug1: Trying private key: /Users/Mike/.ssh/id_ecdsa
debug1: Trying private key: /Users/Mike/.ssh/id_ed25519
debug1: Next authentication method: password
git@<hostname redacted> password:
`

**Comment by RanchoCooper** [comment]
@mike360 Is ur GitLab deployed in Docker like mine ?
my colleague try to install a new GitLab in a ubuntu virtual machine (not in Docker)
it goes well clone by via SSH, but the SMTP can't sent email   ... say !

I am so blue

**Comment by mike360** [comment]
@RanchoCooper Yes, it's in Docker. I'm sure I could get it working outside of Docker but I have a lot of services running on that box, all Dockerized, so I'd rather keep it that way for GitLab too.

**Comment by RanchoCooper** [comment]
the Docker environment was deployed by another one, but he go out for days
I wonder have you set SSH port like this ?   from [Quick Start](https://github.com/sameersbn/docker-gitlab#quick-start)
`
docker run --name gitlab -d \
    --link gitlab-postgresql:postgresql --link gitlab-redis:redisio \
    --publish 10022:22 --publish 10080:80 \
    --env 'GITLAB_PORT=10080' --env 'GITLAB_SSH_PORT=10022' \
    --env 'GITLAB_SECRETS_DB_KEY_BASE=long-and-random-alpha-numeric-string' \
    --volume /srv/docker/gitlab/gitlab:/home/git/data \
    sameersbn/gitlab:8.10.0
`

**Comment by mahzoun** [comment]
@mike360 there is something wrong with your ssh-key, check ssh keys. the ssh connection to your container is ok. and you should be able to clone via ssh after adding your ssh key.
